vs.language.puppet/puppetDef.js

```diff
@@ -263,8 +263,8 @@ define(["require", "exports"], function (require, exports) {
             // dstring.<kind>.<delim> where kind is 'd' (double quoted), 'x' (command), or 's' (symbol)
             // and delim is the ending delimiter (" or `)
             dstring: [
-                [/[^\\`"#]+/, 'string.$S2'],
-                [/#/, 'string.$S2.escape', '@interpolated'],
+                [/[^\\`"\$]+/, 'string.$S2'],
+                [/\$/, 'string.$S2.escape', '@interpolated'],
                 [/\\$/, 'string.$S2.escape'],

```

Comment:
It is a double escape, so if you have a litteral '\$' instead of the regex \$ for example "\\$" would be met as a match for line 268 and not '\$'  and thus it would not fall into the '@interpolated' parsing documented lower